pkg/registry/core/service/allocator/bitmap.go

```diff
@@ -78,6 +108,10 @@ func (r *AllocationBitmap) Allocate(offset int) (bool, error) {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
+	// max is the maximum size of the usable items in the range
+	if offset < 0 || offset >= r.max {
+		return false, fmt.Errorf("the provided offset %d does not match the current range", offset)
+	}
```

Comment:
we had this off by one bug in the bitmap, but the ipallocator and portallocator that consumes this bitmap do their own checks, so the bug it is only local to this package

Reply:
wrong, there was no bug , we didn't check the offset but that was not the problem
I've introduced the off by one bug in the new strategy I've added , and since this didn't check the boundaries :shrug: